server/src/modules/maps/maps.service.ts

```diff
@@ -440,6 +441,29 @@ export class MapsService {
 
     //#endregion
 
+    async getRanks(mapID: number, query: MapRanksGetQuery): Promise<PaginatedResponseDto<MapRankDto>> {
+        const map = await this.mapRepo.get(mapID);
+
+        if (!map) throw new NotFoundException('Map not found');
+
+        const where: Prisma.UserMapRankWhereInput = {
+            mapID: mapID,
+            flags: query.flags
+        };
+
+        if (query.playerID) where.userID = query.playerID;
+        if (query.playerIDs) where.userID = { in: query.playerIDs };
+
+        const order: Prisma.UserMapRankOrderByWithAggregationInput = {};
+        if (query.orderByDate !== undefined) {
```

Comment:
You don't need to check for `undefined` specifically here, validation will have sanitised this. Just ` if (query.orderByDate)` is fine.